### PR TITLE
fix: distinguish between panic-errors and possible-fix-errors

### LINKED ISSUE #464: Remove panicking errors in manta-pay code
remove `unwrap` in manta-pay code, this requires some interface changes of upstream code, i.e. `manta-accounting`

### PR DESCRIPTION
**Pull request by ferrell-code**
Signed-off-by: Charles Ferrell <charlie@manta.network>

closes: #464

### CODE REVIEW

**Review by Garandor — COMMENTED** [review]
looks fine so far

**Review by Garandor — APPROVED** [review]
OK so if I get this correctly, the extrinsic will simply fail if we encounter an `InternalLedgerError`.

We need a followup topic to this:
Trigger tx-pause freeze of chain state on `InternalLedgerError`

File: pallets/manta-pay/src/lib.rs [hunk 1]
```diff
@@ -767,102 +788,172 @@ impl<L, R> AsRef<R> for WrapPair<L, R> {
...
     #[inline]
-    fn is_not_registered(&self, utxo: config::Utxo) -> Option<Self::ValidUtxo> {
-        if UtxoSet::<T>::contains_key(Utxo::try_from(utxo).ok()?) {
-            None
+    fn is_not_registered(
+        &self,
+        utxo: config::Utxo,
+    ) -> Result<Option<Self::ValidUtxo>, Self::Error> {
+        if UtxoSet::<T>::contains_key(
+            Utxo::try_from(utxo).map_err(ReceiverLedgerError::UtxoDecodeError)?,
```
Comment (ghzlatarev) [inline pallets/manta-pay/src/lib.rs:929]: This is not really a decode error. The utxo can simply be missing.

Comment (ferrell-code) [reply pallets/manta-pay/src/lib.rs:929]: If Utxo is missing from storage then it returns `Ok(None)`

Which goes back to your the comment on that maybe should be an error

Comment (ghzlatarev) [reply pallets/manta-pay/src/lib.rs:929]: oh wait my bad, this is the try_from error

**Review by ghzlatarev — APPROVED** [review]

**Pull request merged**